src/prefect/cli/deploy.py

```diff
@@ -390,6 +392,16 @@ async def _run_single_deploy(
                 )
         flow_name = flow.name
     elif flow_name:
+        app.console.print(
+            generate_deprecation_message(
+                "The ability to deploy by flow name",
+                end_date="Jun 2023",
+                help=(
+                    "\nUse `prefect deploy ./path/to/file.py:flow_fn_name` to specify"
```

Comment:
Hm. Technically we could do even better here if we wanted. We could suggest the _actual_ path and flow name to use if it's present in the file i.e. from `entrypoint = flows[flow_name]`

Reply:
Not sure if you feel like that's worth it

Reply:
That would be cool if we were doing a 6-month deprecation but since it's just a couple weeks I don't know that it's worth it.

Reply:
Projects are currently still in beta 🤷‍♀️ so they're subject to breaking changes

Reply:
Hm there's no experimental or beta warning on the `prefect deploy` command — we should be careful about that in the future. Anyway it definitely doesn't make sense to polish it for a 2 week period.

Reply:
👍